addons/sale_project/models/project.py

```diff
@@ -66,6 +66,7 @@ def action_create_invoice(self):
             if task.sale_order_id.id not in sols:
                 sols += [task.sale_order_id.id]
         action['context'] = {
+            'active_id': self.sale_order_id.id if len(sols) == 1 else False,
```

Comment:
My last comment was not just to change the condition but to know what we want because I think we could have a SO from task and not from the current project. If it is just the one in the project then ok, we could use the `self` in the condition too. But if we want the SO in either the current project or in the related tasks then we need to use `sols`. 🤔 

By the way, the variable name is not correct since it contains ids of SO and not ids of SOL. 😕
I think we could rename it into `so_ids`. Moreover, we could do this:
```diff
+ so_ids = set(self.sale_order_id.ids) | set(self.task_ids.sale_order_id.ids)
- sols = self.sale_order_id.ids
- for task in self.task_ids:
-     if task.sale_order_id.id not in sols:
-         sols += [task.sale_order_id.id]
```
(sorry I didn't tell you earlier... 😢 )

What do you think ? 🙂
If it is ok for you, then we can do this since a set it is not subscriptable as a list: 
```suggestion
            'active_id': so_ids.pop() if len(so_ids) == 1 else False,
```

If you change as I propose in this comment then we also need to test if the active_ids need a list or if we can give a set.

Reply:
I agree with your first comment as the compute _compute_has_any_so_to_invoice considers both SO linked to the project itself or to the tasks of the project, so it would make sense to use ```len(sols)``` instead of ```len(self)``` as you suggested.

I do apparently get an error when I try to use a set as an active_id: ```psycopg2.errors.InvalidTextRepresentation: invalid input syntax for type integer: "set()" LINE 1: ...order"."company_id" in (1)) AND "sale_order".id IN ('set()')```. Do you then agree with the way I currently have the code?